### PR TITLE
Adding creosote and steam back

### PR DESCRIPTION
**Pull request by liach**
At the same time, I also added recipes that support fluids and cleaned up some deprecated methods.

### CODE REVIEW

**Comment by liach** [comment]
Should I do #1106 at the same time?

**Comment by liach** [comment]
How about #309? Do I add that too?
By the way, I've fixed #1207

File: src/main/java/mods/railcraft/common/carts/ItemLocoElectric.java [hunk 1]
```diff
@@ -33,12 +33,10 @@ public ItemLocoElectric(IRailcraftCartContainer cart) {
     }
 
     @Override
-    public void finalizeDefinition() {
-        super.finalizeDefinition();
+    public void defineRecipes() {
+        super.defineRecipes();
         //TODO: Change to battery
```
Comment (CovertJaguar) [inline src/main/java/mods/railcraft/common/carts/ItemLocoElectric.java]: This TODO can be removed.

File: src/main/java/mods/railcraft/common/util/crafting/ShapelessFluidRecipe.java [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+     *
+     * @return The recipes input vales.
+     */
+    public List<Object> getInput() {
+        return this.input;
+    }
+
+    @Override
+    public ItemStack[] getRemainingItems(InventoryCrafting inv) {
+        return ForgeHooks.defaultRecipeGetRemainingItems(inv);
```
Comment (CovertJaguar) [inline src/main/java/mods/railcraft/common/util/crafting/ShapelessFluidRecipe.java]: If this is copied directly from my PR, then this part is incomplete. It needs to find and drain the correct amount from the containers.

File: src/main/java/mods/railcraft/common/blocks/RailcraftBlocks.java [hunk 3]
```diff
@@ -206,23 +191,37 @@ public void register() {
             if (!(block instanceof IRailcraftBlock))
                 throw new RuntimeException("Railcraft Blocks must implement IRailcraftBlock");
             IRailcraftBlock blockObject = (IRailcraftBlock) block;
-            blockObject.initializeDefinintion();
-            Railcraft.instance.recipeWaitList.add(blockObject);
```
Comment (CovertJaguar) [inline src/main/java/mods/railcraft/common/blocks/RailcraftBlocks.java:207]: Where did the recipe definitions go?

**Review by CovertJaguar — COMMENTED** [review]
What does the Creosote potion effects do?

File: src/main/java/mods/railcraft/common/blocks/RailcraftBlocks.java (hunk 3 above)
Comment (liach) [reply src/main/java/mods/railcraft/common/blocks/RailcraftBlocks.java:207]: It goes to `RailcraftModulePayload`, in one of the initializing methods.

File: src/main/java/mods/railcraft/common/util/crafting/ShapelessFluidRecipe.java (hunk 2 above)
Comment (liach) [reply src/main/java/mods/railcraft/common/util/crafting/ShapelessFluidRecipe.java]: Will adjust that. Or can I keep a list after the `getResult` call, presuming that this is always called after getting the result?

File: src/main/java/mods/railcraft/common/carts/ItemLocoElectric.java (hunk 1 above)
Comment (liach) [reply src/main/java/mods/railcraft/common/carts/ItemLocoElectric.java]: :+1:

**Comment by liach** [comment]
As creosote is supposed to kill bugs for the wooden ties, the creosote potion effect deals damage to the arthropods with an efficiency similar to that of poison potion to players. It can kill arthropods, however. Non-arthropods will not be affected, but if an arthropod attacks you, it will get half of the length of the creosote effect on you, if there is any. When living entities stay in creosote liquid, they will get 5 seconds of creosote, which will reapply once it expires. You can put creosote bottle as an ingredient to turn mundane potion into creosote potion. It has a long version and a strong version, similar to those of the speed effect.

**Comment by liach** [comment]
@CovertJaguar Please review again,,, Thanks!

File: src/main/java/mods/railcraft/common/util/crafting/ShapelessFluidRecipe.java (hunk 2 above)
Comment (liach) [reply src/main/java/mods/railcraft/common/util/crafting/ShapelessFluidRecipe.java]: Done. I checked in `matches` and used that drain results.

**Review by CovertJaguar — APPROVED** [review]

**Comment by liach** [comment]
I will include this for 10.2.1 for it has a bugfix for fluid container recipes.

**Pull request merged**